src/osyris/spatial/subdomain.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/osyris-project/osyris)
+
+import numpy as np
+
+
+def extract_sphere(dataset, radius, origin):
+    """
+    Extract a spherical subdomain around an origin point.
+    """
+    subdomain = dataset.__class__(nout=dataset.meta["nout"], path=dataset.meta["path"])
+
+    for name, group in dataset.items():
+        pos = group.get("position", group.parent["amr"]["position"])
```

Comment:
If we leave out the change with adding a `.kind` property to datagroups to ensure we are reading the right positions, you could maybe verify that the shape of `pos` is the same as `group.shape`? Because below, you could have a weird case where the indexing into the numpy arrays still work, even if the initial position and other variable arrays had different lengths.

Reply:
> Because below, you could have a weird case where the indexing into the numpy arrays still work, even if the initial position and other variable arrays had different lengths.

The only case I can think of where this could happen is if the user has created a group with no position vector and is non-amr dependent, in which case the code still grabs the amr positions and tries to index arrays with different lengths. What do we do in this case? Throw an error? Or simply ignore that group?

Reply:
> What do we do in this case? Throw an error? Or simply ignore that group?

This is why I proposed the solution with adding a `.kind` property to the Datagroup. Without that, I don't know what you can do. But because you said you prefered delaying this to a later change, I'm saying you should at least have a check that the arrays have the same length.

Users are free to create datagroups, so yes, we have to keep this possibility in mind, and not rely on the hope that users won't mess thing up (because they most definitely will, if given the possibility).